### PR TITLE
spike: figma + storybook

### PR DESCRIPTION
**Pull request by DarioR01**
Resolves JIRA [WSTEAM1-357](https://jira.dev.bbc.co.uk/browse/WSTEAM1-357)

Let me know if the documentation is ok and if I am missing anything. Maybe we can add more ruleset to follow.
Also, what do we think of Figma previews? This is still a spike and won't be pushed unless the team likes the idea and is willing to maintain this.

considerations:
- Is this gonna help us find Designs?
- Is this too much extra work for devs?
- Should we add this in the definition od done? Is this part of it?

Feel free to just comment the PR without approval, anyone feedback is welcome :)

Documentation: https://5d28eb3fe163f6002046d6fa-pbeguainwl.chromatic.com/?path=/story/coding-standards-storybook--page#how-to-add-figma-previews

Figma addon example: 
https://5d28eb3fe163f6002046d6fa-slittjmwcn.chromatic.com/?path=/story/containers-front-page-story-row--with-top-row
Screenshot 2023-03-27 at 13 08 21
This example cannot be found in this PR anymore as it was removed.

Overall changes
======
- Added code standards documentation
- Added `storybook-addon-designs` in devDependencies for Figma preview in storybook stories. Example of this showed in the newly created documentation
Code changes
- Added Global storybook providers: Now all providers are automatically added to our components in storybook. To switch services use the globe icon:
Screenshot 2023-04-06 at 14 01 47

======

- _A bullet point list of key code changes that have been made._

Testing
======
N/A

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

### CODE REVIEW

File: docs/Coding-Standards/storybook.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,415 @@
...
+};
+
+// Export an arrow function with the variants of your components
+export const Variant1 = () => <MyComponent prop1={variant1} />;
+
+// The name declaration of this function will also be the name that storybook will give to this variant
+export const Variant2 = () => <MyComponent prop1={variant2} />;
+```
+
+### Creating story for a component that require providers
```
Comment (HarveyPeachey) [inline docs/Coding-Standards/storybook.stories.mdx]: Unrelated to this ticket, could we use the [global decorator](https://storybook.js.org/docs/react/writing-stories/decorators#global-decorators) to handle all of our providers, so we don't have to include them individually for each story that needs them? Similar to our react-testing-libray-with-providers [setup](https://github.com/bbc/simorgh/blob/latest/src/app/components/react-testing-library-with-providers.tsx)

**Review by HarveyPeachey — COMMENTED** [review]
Looks really good. Just needs a lil but of proof reading, but other than that really like the structure of the document. 

The figma integration looks good, I like the idea of colocation, as long as the designs have a confirmed layout, like breakpoint grouping used, it should be relatively easy and quick for devs to create the config to display those designs. It also makes it easier for us if we need to refer to designs in the past, as we can access them quite quickly.

It looks like there is a lot of boilerplate/metadata needed to generate a fully fledged story, maybe worth off the back of this creating a template story, or creating a script that walks through what options you want for the story, and depending on the options you pick creates a nice template for the story, not necessary but would a be cool nice to have.

I've also added a comment about providers, which is out of the scope of this spike

**Comment by greenc05** [comment]
This looks great @DarioR01 

I would suggest we do a recommended ordering for the ‘tabs’. Starting with the guts:

1) Screen reader UX
2) Typography
3) States
4) Groups…

One other suggestion is that we could add to the ‘Heath Factors’ component is a check for this: ‘Design documentation’
Screenshot 2023-03-29 at 15 27 35

**Review by Isabella-Mitchell — COMMENTED** [review]
I agree I like the co-location - and good work. Thanks for looking into this!

I personally find the 'Group 4' design a bit hard to make sense of. All the other designs appear to have quite a defined 'canvas', but the Group 4 design just looks a bit plonked in the middle of the screen. Though maybe that's just me  🙃

Do we know how much chance there is for these URLs changing/ getting broken? I assume since Figma/ storybook encourage linking by URL that they are pretty concrete.

I can imagine this being most useful for keeping track of figma files (since that was one of our biggest complaints about Zeplin). As well as useful for the side by side comparison. So even though it would mean a little bit more work for devs when they're creating a new component, it would make revisiting that component a lot easier in the future.

File: docs/Coding-Standards/storybook.stories.mdx (hunk 1 above)
Comment (DarioR01) [reply docs/Coding-Standards/storybook.stories.mdx]: > It looks like there is a lot of boilerplate/metadata needed to generate a fully fledged story, maybe worth off the back of this creating a template story, or creating a script that walks through what options you want for the story, and depending on the options you pick creates a nice template for the story, not necessary but would a be cool nice to have.

I see potential for another mini project 😍 

> Unrelated to this ticket, could we use the [global decorator](https://storybook.js.org/docs/react/writing-stories/decorators#global-decorators) to handle all of our providers, so we don't have to include them individually for each story that needs them? Similar to our react-testing-libray-with-providers [setup](https://github.com/bbc/simorgh/blob/latest/src/app/components/react-testing-library-with-providers.tsx)

Theoretically yes, In practice stories are very dynamic and single components have stories in multiple services. So either we have something like this:
```jsx
const componentWithProviders = ({service,...props}) => {
    <AllTheProviders service={service}>  
        <Component ...props/>
    </AllTheProviders>
}
```

or I look into dynamic decorators:

```jsx
export default {
  title: 'Component',
  Component,
  decorators: [AllTheProviders], // This needs to be dynamic.
  parameters: {
    ...
  },
};
```

**Comment by DarioR01** [comment]
> 

I agree on the ordering, but we are already using the tabs to link the Design Files.
The only benefit I see on adding that is to know if the designs have been added or not.

**Comment by DarioR01** [comment]
> I agree I like the co-location - and good work. Thanks for looking into this!
> 
> I personally find the 'Group 4' design a bit hard to make sense of. All the other designs appear to have quite a defined 'canvas', but the Group 4 design just looks a bit plonked in the middle of the screen. Though maybe that's just me 🙃
> 
> Do we know how much chance there is for these URLs changing/ getting broken? I assume since Figma/ storybook encourage linking by URL that they are pretty concrete.
> 
> I can imagine this being most useful for keeping track of figma files (since that was one of our biggest complaints about Zeplin). As well as useful for the side by side comparison. So even though it would mean a little bit more work for devs when they're creating a new component, it would make revisiting that component a lot easier in the future.

Single links should not break unless UX deletes the section the link was pointing at. This is not necessarily bad since it means that the design has been modified and therefore out of date.

File: docs/Coding-Standards/storybook.stories.mdx (hunk 1 above)
Comment (DarioR01) [reply docs/Coding-Standards/storybook.stories.mdx]: Alright, a global providers proposal for storybook has just been pushed. Let me know what you think about it.

**Review by HarveyPeachey — APPROVED** [review]
The new global provider setup looks good to me

File: .storybook/preview.js [hunk 2]
```diff
@@ -318,6 +324,253 @@ const theme = create({
...
+  },
+};
+
+addDecorator((Story, context) => {
+  return (
+    <ThemeProvider service={context.globals.service.service}>
+      <ToggleContextProvider toggles={{}}>
+        <ServiceContextProvider
+          service={context.globals.service.service}
+          variant={context.globals.service.variatn}
```
Comment (karinathomasbbc) [inline .storybook/preview.js]: Typo
```suggestion
          variant={context.globals.service.variant}
```

**Review by karinathomasbbc — APPROVED** [review]
Looks good, let's trial it and see how it goes!

**Pull request merged**